agent/tcs/client/client_test.go

```diff
@@ -309,6 +379,51 @@ func TestPublishOnceNonIdleStatsEngine(t *testing.T) {
 	}
 }
 
+func TestPublishServiceConnectStatsEngine(t *testing.T) {
+	expectedRequests := 3
+	numTasks := 11
+	cs := clientServer{
+		statsEngine: newServiceConnectStatsEngine(numTasks),
+	}
+
+	tempLimit := publishMetricRequestSizeLimit
+	publishMetricRequestSizeLimit = testPublishMetricRequestSizeLimit
+	defer func() {
+		publishMetricRequestSizeLimit = tempLimit
+	}()
+
+	requests, err := cs.metricsToPublishMetricRequests(true)
+	if err != nil {
+		t.Fatal("Error creating publishmetricrequests: ", err)
+	}
+
+	taskArns := make(map[string]bool)
+	for _, request := range requests {
+		for _, taskMetric := range request.TaskMetrics {
+			_, exists := taskArns[*taskMetric.TaskArn]
+			if exists {
```

Comment:
This was just to test the part of task metric being sent in next message has empty `ContainerMetrics` since we don't want agent to send it more than once to TACS.

I can add validation to check that  `ContainerMetrics` in the first part of `taskMetric` is not empty.